iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -1277,6 +1277,39 @@ TYPED_TEST(stringTyped_test, StringWithContentIsNotEmtpy)
     EXPECT_THAT(sut.empty(), Eq(false));
 }
 
+/// @note void clear() noexcept
+TYPED_TEST(stringTyped_test, ClearEmptyStringDoesNotChangeString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5109413a-6067-4f7f-ac35-8dd3a33a641f");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString().capacity();
+
+    this->testSubject.clear();
+    EXPECT_THAT(this->testSubject.empty(), Eq(true));
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
+}
+
+TYPED_TEST(stringTyped_test, ClearNotEmptyStringResultsInEmptyStringWithUnchangedCapacity)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d25d2a74-6ea0-4892-b072-02177b31309e");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString().capacity();
+    this->testSubject = "M";
+    ASSERT_THAT(this->testSubject.empty(), Eq(false));
+
+    this->testSubject.clear();
+    EXPECT_THAT(this->testSubject.empty(), Eq(true));
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
+}
+
+TYPED_TEST(stringTyped_test, ChangeStringAfterClearWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "58444d00-ec85-464f-8269-7838823f04c2");
+    this->testSubject.clear();
+    this->testSubject = "M";
```

Comment:
Nitpick: You could think about defining the char string `M` as a constexpr and use it at the two places. Might be seem a bit over the top here, but in generally a good approach if test cases are larger.

Reply:
I totally agree with you. Good approach in general, but in this particular case too much.